### PR TITLE
Allow users to renew books + Allow library managers to view loan history

### PR DESCRIPTION
**Pull request by helenatran**
#### What does this PR do?
-> Allow users to renew borrowed books (US12)
-> Add a loan history section in the show page of a book which includes: borrow date, return date, late return, fine (amount, indicate if pair or unpaid) (US08)

#### How can this PR be tested?
bundle exec rspec

#### Does this PR relate to a User Story? If so, link it here
US08: https://trello.com/c/DKKiMCR5
US12: https://trello.com/c/5QjieHdH

### CODE REVIEW

**Comment by Solacx** [comment]
Seems like there are some changes here taken from PR12. Please make sure requested changes are also reflected here.

**Comment by Solacx** [comment]
Please see Messenger chat regarding this code: image

**Comment by Solacx** [comment]
^ For the above image, please also change the alert to better indicate the error is due to system issue. For example, "Failed to renew book." I believe at the moment it can sound like user is able to fix the issue somehow.

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -75,11 +75,14 @@ def renew
     @book = Book.find(params['book_id'])
     @loan = Loan.find_by(book: @book, user: current_user, returned_at: nil)
     
-    if @loan.update(renewed_at: Time.now)
-      @loan.to_be_returned_at
-      redirect_to @book, notice: 'Book was successfully renewed.'
+    if @loan.to_be_returned_at < Time.now
+      redirect_to @book, alert: 'Book is overdue. Please check your account for late return fees.'
```
Comment (pharapeti) [inline app/controllers/books_controller.rb:79]: Add `and return` to the end of this line. I'll explain later

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -75,11 +75,14 @@ def renew
...
     else
-      redirect_to @book, alert: 'Could not renew book.'
+      if @loan.update(renewed_at: Time.now)
+      redirect_to @book, notice: 'Book was successfully renewed.'
```
Comment (pharapeti) [inline app/controllers/books_controller.rb:82]: lines 82 and 84 need to be tabbed in

**Review by pharapeti — APPROVED** [review]
Looks good 👍

**Pull request merged**